### PR TITLE
Slideshow controls

### PR DESCRIPTION
**Pull request by davidrapson**
Part of https://github.com/guardian/membership-frontend/pull/700 which can go out separately.

- Refactor slideshow JS to allow next/prev controls
- Update welcome page to use refactored slideshow
- Update homepage to use new slideshow for goods shed images

screen shot 2015-08-25 at 15 34 15

@afiore

### CODE REVIEW

File: frontend/assets/javascripts/src/modules/slideshow.js [hunk 1]
```diff
@@ -1,48 +1,88 @@
...
-                    cycleItems(items, el.getAttribute('data-slideshow-duration') || 5000);
-                }
-            }, false);
-        }
+        if (!slideshows.length) { return; }
+        [].forEach.call(slideshows, function(slideshow) {
+            var items = slideshow.querySelectorAll(SLIDESHOW_CHILDREN);
+            if(!items.length) { return; }
+            setCurrentItem(slideshow, items, 0);
+            if(slideshow.getAttribute(AUTOPLAY_ATTR) || false) {
```
Comment (afiore) [inline frontend/assets/javascripts/src/modules/slideshow.js]: is this `|| false` needed?

Comment (davidrapson) [inline frontend/assets/javascripts/src/modules/slideshow.js]: Not essential, but autoplay is optional and I prefer the explicit `false` than relying on the null value of `getAttribute`

**Comment by davidrapson** [comment]
Minor update to show captions

screen shot 2015-08-25 at 16 27 33

**Comment by davidrapson** [comment]
@afiore Look OK otherwise?

**Comment by afiore** [comment]
:+1: Fine on the Scala front. I haven't given look deeply into the JS code but I couldn't spot anything major

**Pull request merged**